spacemeshos/smapp
# Tweak auto updates: check updates hourly + manually

### What's changed
1. Check for updates automatically every hour (by default) instead of 24 hours.
2. Fixed #1224
3. Added a button that triggers a check for updates next to the version ("refresh" icon).
4. Added `--checkInterval` flag to customize the interval between automatic refreshing discovery service (and therefore check for updates). Also useful for tests ;) It accepts value in seconds, which should be greater than `0` otherwise it will use the default interval of 60 minutes.

### How to test
Run PR build with custom discovery and/or custom interval:
```
./Spacemesh --discovery=http://localhost:8000/networks.json --checkInterval=30
```
Then you can change `minSmappRelease` and/or `latestSmappRelease` in your custom `networks.json` file.
You can also run it without `checkInterval` flag and check for updates manually (or wait an hour).
